### ISSUE TITLE
read_csv_batched should return an instance of an iterator

### ISSUE THREAD

**Issue by aburnsy** [pola-rs/polars#13885]
### Description

read_csv_batched returns an instance of BatchedCsvReader, which we then need to call next_batches method on. We call it like so:
```python
reader = pl.read_csv_batched("big_file.csv") 
while (batches :=  reader.next_batches(100)) is not None
    df = pl.concat(batches)
```

Would it be possible to instead return an iterator instance (with __next__ and __iter__ methods)? Doing so would allow us to use a for loop. 

```python
reader = pl.read_csv_batched("big_file.csv", number_of_batches_per_iteration=100) 
for batches in reader: 
    df = pl.concat(batches)
```

I believe this to be more concise and intuitive for users of both Rust and Python.

**Comment by aburnsy** [comment]
I meant to add, in this case we are pushing the number_of_batches_per_iteration param from the next_batches method into the read_csv_batched method.

I appreciate that this would be a breaking change.

**Comment by cmdlineluser** [comment]
@mcrumiller I believe each batch is a partition by row count. (I'm not fully aware of why it works this way.)

There have been a few `read_csv_batched` questions on SO and there does seem to be some confusion with how to use it correctly.

I had been using `iter()` 

```python
for batches in iter(lambda: reader.next_batches(100), None):
    ...
```

**Comment by mcrumiller** [comment]
So each iteration you are grabbing 100 batches at a time? Why not just reduce the size of each batch by a factor of 100? Or does that mean grab the next 100 rows?

**Comment by cmdlineluser** [comment]
Yeah, `.next_batches(N)` returns a list of `N` batches that has `batch_size` rows.

> batch_size: [int](https://docs.python.org/3/library/functions.html#int) = 50000

```python
pl.read_csv_batched(batch_size = 50000)
```

As you are saying, I'm not sure why the multiple batches thing exists.

**Comment by mcrumiller** [comment]
There must be a reason for this. Tracked its origin to #5212. @ritchie46 what's the rationale for retrieving multiple batches at once?

We could implement a python iterator that uses `reader.next_batches(1)` which would allow someone to cleanly iterate through each batch as per the  OP.

**Comment by ritchie46** [comment]
Because the iterator trait forces a single batch. This API is more generic. You can always reduce to a single batch per iteration if you want. But it will be slower and for some readers single threaded.

**Comment by deanm0000** [comment]
As an aside, in answering [ths](https://stackoverflow.com/questions/77593648/how-to-split-100-million-18-5-gb-csv-rows-into-50k-each-small-excel-files/77601125#77601125) question I noticed that the batch size isn't strictly the row size of each batch. I'm not sure if that's intentional or not but just flagging it.

**Comment by mimakaev** [comment]
Fully support returning an iterator. The method as is is very confusing.

**Comment by CarMoreno** [comment]
Check [this issue](https://github.com/pola-rs/polars/issues/19978) @deanm0000 , It is related to your comment.
